cms/djangoapps/contentstore/utils.py

```diff
@@ -296,7 +296,7 @@ def get_group_display_name(user_partitions, xblock_display_name):
     """
     for user_partition in user_partitions:
         for group in user_partition['groups']:
-            if str(group['id']) in xblock_display_name:
+            if _(u'Group ID {group_id}').format(group_id=group['id']) == xblock_display_name:
```

Comment:
I added this because `xblock_display_name` is actually retried from xblock which has same translated version, that is displayed to user (as a translated version), that is why I am checking this.